proposals/ref-struct-interfaces.md

```diff
@@ -116,16 +116,47 @@ T M3<T>(Span<T> span)
 
 These parameters will be encoded in metadata as described in the [byref-like generics doc](https://github.com/dotnet/runtime/blob/main/docs/design/features/byreflike-generics.md). Specifically by using the `gpAcceptByRefLike(0x0020)` attribute value.
 
-Detailed notes: 
-- A `allow T : ref struct` generic parameter cannot 
+Detailed notes:
+- A `allow T : ref struct` generic parameter cannot
     - Have `where T : U` where `U` is a known reference type
     - Have `where T : class` constraint
     - Cannot be used as a generic argument unless the corresponding parameter is also `allow T: ref struct`
 - A type parameter `T` which has `allow T: ref struct` has all the same limitations as a `ref struct` type.
 
+## Soundness
+
+We would like to verify the soundness of both the `ref struct` anti-constraint in particular and the anti-constraint concept in general. To do so we'd like to take advantage of the existing soundness proofs provided for the C# type system. This task is made easier by defining a new language that is similar to C#, but more regular in construction. We will verify the safety of that model, and then specify a sound translation to this language. Because this new language is centered around constraints, we'll call this language "constraint-C#".
+
+The primary ref struct safety invariant that must be preserved is that variables of ref struct type must not appear on the heap. We can encode this restriction via a constraint. Because constraints permit substitution, not forbid it, we will technically define the inverse constraint: `heap`. The `heap` constraint specifies that a type may appear on the heap. In "constraint-C#" all types satisfy the `heap` constraint except for ref-structs. Moreover, all existing type parameters in C# will be lowered to type parameters with the `heap` constraint in "constraint-C#".
+
+Now, assuming that existing C# is safe, we can transfer the C# ref-struct rules to "constraint-C#".
+
+  1. Fields of classes cannot have a ref-struct type.
+  2. Static fields cannot have a ref-struct type.
+  3. Variables of ref-struct type cannot be converted to non-ref structs.
+  4. Variables of ref-struct type cannot be substituted as type arguments.
+  5. Variables of ref-struct type cannot implement interfaces.
+
+The new rules apply to the `heap` constraint:
+
+  1. Fields of classes must have types that satisfy the `heap` constraint.
+  2. Static fields must have types that satisfy the `heap` constraint.
+  3. Variables of types which do not satisfy the `heap` constraint cannot be converted to a type without the `heap` constraint.
```

Comment:
the 'do not', 'cannot' and 'without' are tripping my brain up a lot reading this sentence.  any posibility on simplifying this?

Reply:
I've changed it to something that I believe is equivalent, but I might be missing a case. I would have to double check the spec.